### PR TITLE
Add ONNXRT Parameters class refactor

### PR DESCRIPTION
**Pull request by jafet-chaves**

### CODE REVIEW

File: r2i/onnxrt/parameters.h [hunk 1]
```diff
@@ -40,89 +41,21 @@ class Parameters: public IParameters {
   RuntimeError Set (const std::string &in_parameter, int in_value) override;
   RuntimeError Set (const std::string &in_parameter, double in_value) override;
   RuntimeError List (std::vector<ParameterMeta> &metas) override;
+  /* To set/get Parameters of the internal Engine instance */
+  RuntimeError SetLogLevel (int &value);
```
Comment (michaelgruner) [inline r2i/onnxrt/parameters.h:45]: Setters shouldn't  receive by reference (except for the string, for which I recommend `const std::string`

File: r2i/onnxrt/parameters.h [hunk 2]
```diff
@@ -40,89 +41,21 @@ class Parameters: public IParameters {
...
-
-  class GraphOptLevelAccessor : public IntAccessor {
-   public:
-    GraphOptLevelAccessor (Parameters *target) : IntAccessor(target) {}
-
-    RuntimeError Set () {
-      return target->engine->SetGraphOptLevel(this->value);
-    }
+ protected:
+  std::shared_ptr <IEngine> engine;
```
Comment (michaelgruner) [inline r2i/onnxrt/parameters.h:55]: Why are you saving the interface and not the specific class?

File: r2i/onnxrt_openvino/parameteraccessors.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+#include <r2i/runtimeerror.h>
+#include <r2i/iparameters.h>
+
+namespace r2i {
+namespace onnxrt_openvino {
+
+class HardwareIdAccessor : public r2i::onnxrt::StringAccessor {
+ public:
+  HardwareIdAccessor () {}
```
Comment (michaelgruner) [inline r2i/onnxrt_openvino/parameteraccessors.h:23]: Unneeded

File: tests/unit/r2i/onnxrt/parameters.cc [hunk 4]
```diff
@@ -17,31 +17,6 @@
 #include <CppUTest/CommandLineTestRunner.h>
 #include <CppUTest/TestHarness.h>
 
-namespace mock {
```
Comment (michaelgruner) [inline tests/unit/r2i/onnxrt/parameters.cc:20]: Why did you remove all these?

File: tests/unit/r2i/onnxrt_openvino/parameters.cc [hunk 5]
```diff
@@ -18,287 +18,40 @@
...
-  error = parameters.Configure(engine, model);
-  LONGS_EQUAL (r2i::RuntimeError::Code::EOK, error.GetCode ());
-
-  error = parameters.Get("undefined-parameter", value);
-  LONGS_EQUAL (r2i::RuntimeError::Code::INVALID_FRAMEWORK_PARAMETER,
-               error.GetCode ());
-}
-
-TEST (OnnxrtOpenVinoParameters, SetAndGetLogId) {
+TEST (OnnxrtOpenVinoParameters, SetHardwareIdAtWrongEngineState) {
```
Comment (michaelgruner) [inline tests/unit/r2i/onnxrt_openvino/parameters.cc:36]: Sees like we are missing a bunch of tests. Wrong values, wrong types, unknown parameters, etc...

File: r2i/onnxrt/parameters.h (hunk 1 above)
Comment (jafet-chaves) [reply r2i/onnxrt/parameters.h]: Solved [here](https://github.com/RidgeRun/r2inference/commit/f60ee705f623ed661e84f61aa9f3b3a541adcc09)

File: r2i/onnxrt/parameteraccessors.cc [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+  RuntimeError error;
+
+  if (nullptr == target) {
+    error.Set (r2i::RuntimeError::Code::NULL_PARAMETER,
+               "Trying to access null IParameters pointer");
+    return error;
+  }
+
+  r2i::onnxrt::Parameters *downcast_parameters =
+    dynamic_cast<r2i::onnxrt::Parameters *>(target);
```
[the remark this answers is not in the document]
Comment (jafet-chaves) [reply r2i/onnxrt/parameteraccessors.cc:29]: Solved [here](https://github.com/RidgeRun/r2inference/commit/c2f106e82ae2dc117fdc87c7031326a0da8fbfd6)

File: r2i/onnxrt/parameters.cc [hunk 7]
```diff
@@ -59,22 +59,8 @@ RuntimeError Parameters::Configure (std::shared_ptr<r2i::IEngine> in_engine,
...
-  auto model = std::dynamic_pointer_cast<Model, IModel>(in_model);
-  if (nullptr == model) {
-    error.Set (RuntimeError::Code::INCOMPATIBLE_MODEL,
-               "The provided engine is not an ONNXRT model");
-    return error;
-  }
-
-  this->engine = engine;
-  this->model = model;
+  this->engine = in_engine;
```
[the remark this answers is not in the document]
Comment (jafet-chaves) [reply r2i/onnxrt/parameters.cc]: Solved [here](https://github.com/RidgeRun/r2inference/commit/63f9657aa4f24652bc8298afad787bf87379eaa5)

File: r2i/onnxrt/parameters.cc [hunk 8]
```diff
@@ -212,5 +183,89 @@ RuntimeError Parameters::List (std::vector<ParameterMeta> &metas) {
...
+               "Parameter \"" + in_parameter + "\" is not of type " + stype);
+    return undefined;
+  }
+
+  return param;
+}
+
+RuntimeError Parameters::SetLogLevel (int &value) {
+  RuntimeError error;
+  auto downcast_engine = std::dynamic_pointer_cast<Engine, IEngine>(this->engine);
```
[the remark this answers is not in the document]
Comment (jafet-chaves) [reply r2i/onnxrt/parameters.cc]: Solved [here](https://github.com/RidgeRun/r2inference/commit/63f9657aa4f24652bc8298afad787bf87379eaa5)

File: r2i/onnxrt/parameters.h (hunk 2 above)
Comment (jafet-chaves) [reply r2i/onnxrt/parameters.h]: Solved [here](https://github.com/RidgeRun/r2inference/commit/63f9657aa4f24652bc8298afad787bf87379eaa5)

File: tests/unit/r2i/onnxrt/parameters.cc (hunk 4 above)
Comment (jafet-chaves) [reply tests/unit/r2i/onnxrt/parameters.cc]: Solved [here](https://github.com/RidgeRun/r2inference/commit/481b2ed3b0e12d38d57cd863cd25238d368d85f0)

File: tests/unit/r2i/onnxrt_openvino/parameters.cc (hunk 5 above)
Comment (jafet-chaves) [reply tests/unit/r2i/onnxrt_openvino/parameters.cc:122]: Solved [here](https://github.com/RidgeRun/r2inference/commit/481b2ed3b0e12d38d57cd863cd25238d368d85f0)

File: r2i/onnxrt/parameters.cc [hunk 9]
```diff
@@ -19,27 +19,27 @@ namespace onnxrt {
 Parameters::Parameters () {
   ParamDesc logging_level_desc = {
     {logging_level_meta},
-    std::make_shared<LoggingLevelAccessor>(this)
+    std::make_shared<r2i::onnxrt::LoggingLevelAccessor>()
```
[the remark this answers is not in the document]
Comment (jafet-chaves) [reply r2i/onnxrt/parameters.cc:22]: Returning the list in the constructor limits the possibility to expand the unordered_map in the constructor of the OpenVINO Parameters derived class.

File: r2i/onnxrt/parameteraccessors.cc [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+*/
+
+#include "parameters.h"
+#include "parameteraccessors.h"
+
+namespace r2i {
+namespace onnxrt {
+Accessor::Accessor () {}
```
[the remark this answers is not in the document]
Comment (jafet-chaves) [reply r2i/onnxrt/parameteraccessors.cc:17]: To have a common object type inside the struct ParamDesc which ultimately is stored in the unordered_map.

[the remark this answers is not in the document]
Comment (jafet-chaves) [reply r2i/onnxrt/parameteraccessors.cc]: Solved [here](https://github.com/RidgeRun/r2inference/commit/c894381c1d7e0655b6c6031b9888ab9918c4d52b)

File: r2i/onnxrt/parameteraccessors.h [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+ public:
+  Accessor();
+  virtual ~Accessor() {}
+  virtual RuntimeError Set(IParameters *target) = 0;
+  virtual RuntimeError Get(IParameters *target) = 0;
+};
+
+class IntAccessor : public Accessor {
+ public:
+  IntAccessor () : Accessor() {}
```
[the remark this answers is not in the document]
Comment (jafet-chaves) [reply r2i/onnxrt/parameteraccessors.h]: Solved [here](https://github.com/RidgeRun/r2inference/commit/c894381c1d7e0655b6c6031b9888ab9918c4d52b)

File: r2i/onnxrt_openvino/parameteraccessors.h (hunk 3 above)
Comment (jafet-chaves) [reply r2i/onnxrt_openvino/parameteraccessors.h]: Solved [here](https://github.com/RidgeRun/r2inference/commit/c894381c1d7e0655b6c6031b9888ab9918c4d52b)

File: r2i/onnxrt/parameteraccessors.cc [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+
+RuntimeError LoggingLevelAccessor::Set (IParameters *target) {
```
[the remark this answers is not in the document]
Comment (jafet-chaves) [reply r2i/onnxrt/parameteraccessors.cc]: Solved [here](https://github.com/RidgeRun/r2inference/commit/ce51dbd2cc4911319903103ac0db49311f10b109)
